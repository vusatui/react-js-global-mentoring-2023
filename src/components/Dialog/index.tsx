import {ReactNode} from "react";
import {Portal} from "react-portal";

import {
    StyledCloseButton,
    StyledCloseIcon,
    StyledDialogContainer,
    StyledDialogContent,
    StyledDialogHeader, StyledDialogOverlay
} from "./styled";

interface DialogProps {
    children: ReactNode
    isOpened: boolean
    onCloseButtonClick: () => void
}

const Dialog = ({
    children,
    isOpened,
    onCloseButtonClick,
}: DialogProps) => (
    <>
        {isOpened && <Portal>
            <StyledDialogOverlay>
                <StyledDialogContainer>
                    <StyledDialogHeader>
                        <StyledCloseButton onClick={onCloseButtonClick}>
                            <StyledCloseIcon />
                        </StyledCloseButton>
                    </StyledDialogHeader>
                    <StyledDialogContent>
                        {children}
                    </StyledDialogContent>
                </StyledDialogContainer>
            </StyledDialogOverlay>
        </Portal>}
    </>
);

export default Dialog;