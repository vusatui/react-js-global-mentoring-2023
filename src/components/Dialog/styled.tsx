"use client";

import styled from "styled-components";

import CloseIcon from "../icons/CloseIcon";

export const StyledDialogOverlay = styled.div`
  position: fixed;
  display: flex;
  flex-flow: column;
  align-items: center;
  padding: 157px 0;
  top: 0;
  bottom: 0;
  left: 0;
  right: 0;
  background: rgba(35, 35, 35, 0.918051);
  mix-blend-mode: normal;
  backdrop-filter: blur(12.2323px);
  overflow: auto;
  z-index: 3;
`;

export const StyledDialogContainer = styled.div`
  background: #232323;
  box-shadow: 
          0 0 2px rgba(0, 0, 0, 0.1), 
          0 2px 10px rgba(0, 0, 0, 0.07), 
          0 10px 20px rgba(0, 0, 0, 0.05), 
          0 10px 50px rgba(0, 0, 0, 0.05);
`;

export const StyledDialogHeader  = styled.div`
  display: flex;
  flex-flow: row nowrap;
  justify-content: flex-end;
  align-items: center;
  padding: 30px 30px 11px 30px;
`;

export const StyledCloseButton = styled.button.attrs({ "data-testid": "close-modal-button" })`
  background: none;
  box-shadow: none;
  padding: 0;
  border: none;
  cursor: pointer;
`;

export const StyledCloseIcon = styled(CloseIcon)`
  width: 20.77px;
  height: 20.77px;
`;

export const StyledDialogContent  = styled.div`
  padding: 0 60px 60px 60px;
`;