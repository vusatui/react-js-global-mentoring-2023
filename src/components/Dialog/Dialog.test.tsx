import {render, screen} from "@testing-library/react";

import {useState} from "react";

import Dialog from "./index";

import userEvent from "@testing-library/user-event";

import Mock = jest.Mock;

let handleCloseButtonClick: Mock;

const Component = () => {
    const [isOpened, setIsOpened] = useState(true);
    handleCloseButtonClick = jest.fn(() => setIsOpened(false));

    return (
        <Dialog
            isOpened={isOpened}
            onCloseButtonClick={handleCloseButtonClick}
        >
            Dialog Content
        </Dialog>
    );
};

describe("Dialog", () => {

    it("Should renders correctly", () => {
        const { asFragment } = render(<Component/>);
        expect(asFragment()).toMatchSnapshot();
        userEvent.click(screen.getByTestId("close-modal-button"));
        expect(handleCloseButtonClick).toBeCalledTimes(1);
        expect(asFragment()).toMatchSnapshot();
    });

});