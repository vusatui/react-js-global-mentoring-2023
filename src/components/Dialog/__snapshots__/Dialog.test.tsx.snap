// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Dialog Should renders correctly 1`] = `<DocumentFragment />`;

exports[`Dialog Should renders correctly 2`] = `<DocumentFragment />`;
